runtime/io.c

```diff
@@ -690,6 +690,7 @@ CAMLprim value caml_channel_descriptor(value vchannel)
 
 CAMLprim value caml_ml_close_channel(value vchannel)
 {
+  CAMLparam1 (vchannel);
```

Comment:
I think it’s worth a comment about the need for the root, given how subtle the trigger is!

Reply:
Your proposal works well within the mindset that registering roots should be avoided unless necessary -- here we cannot avoid it for a subtle reason, so we should document it.

I think that we should think differently, and reflect this in our documentation practices:  we should not require justification for registering values (the easy and safe thing), we should require justification for *not* registering values (the subtle and unsafe thing).